src/lib/y2storage/proposal/autoinst_nfs_planner.rb

```diff
@@ -57,11 +57,25 @@ def planned_devices(drive)
 
       # Similar to {OLD_FORMAT_MANDATORY_VALUES}, but for the new AutoYaST style.
       #
-      # TODO
-      NEW_FORMAT_MANDATORY_VALUES = {}.freeze
+      # With new format, drive section must contain a device and partition section must contain
+      # a mount value, e.g.:
+      #
+      # <drive>
+      #   <device>192.168.56.1:/root_fs</device>
+      #   <partitions>
+      #     <partition>
+      #       <mount>/</mount>
+      #     </partition>
+      #   </partitions>
+      # </drive>
+      NEW_FORMAT_MANDATORY_VALUES = { drive: [:device], partition: [:mount] }.freeze
 
       private_constant :OLD_FORMAT_MANDATORY_VALUES, :NEW_FORMAT_MANDATORY_VALUES
 
+      def old_format?(drive)
+        drive.device == "/dev/nfs" || drive.partitions.any? { |p| !p.device.nil? }
```

Comment:
np: You could write `drive.partitions.any?(&:device)` or even `drive.partitions.any? { |p| p.device }`.